Automattic/wp-calypso
# build(i18n): Drop target support for languages

#### Background

Now that we have dropped the `fallback` build, it doesn't make sense to generate language files in `public/evergreen/languages` and `public/fallback/languages`. It doesn't even make sense to have a `target` anymore.

This PR removes all notion of `target` form the build-languages scripts and generated files.

#### Changes proposed in this Pull Request

* Move translations from `public/<target>/languages` to just `public/languages`
* Move `./calypso-strings.pot` to `./public/calypso-strings.pot`. I think it is better to keep all build artifacts inside `public` and don't pollute the root.
* Move `./chunks-map.<target>.json` to `./public/chunks-map.json`. Ditto.
* Refactor `./bin/build-languages.js` to use `node-fetch` instead of https streams. Streams were getting me lots of `ECONNRESET` problems, and now that we don't need to save a downloaded file in several places, they don't provide any huge benefit over the Fetch API. 
* Refactor `./bin/build-languages.js` to just use `console.log` instead of readline to manipulate stdout. The readline logic was messing with debugging in VSCode and made the build logs hard to follow in Docker.

#### Testing instructions

* Open calypso.live link, go to your profile and change the language. You should see a bunch of `<locale>-.json` files loaded in the Network tab every time you go to a different page.
* Go to gutenboarding (add `/new` to the calypso.live link) and check you can change languages using the top-right lang selector.
